wrapper/src/test/java/software/amazon/jdbc/plugin/readwritesplitting/ReadWriteSplittingPluginTest.java

```diff
@@ -473,4 +474,68 @@ public void testConnect_errorUpdatingHostSpec() throws SQLException {
             this.mockConnectFunc));
     verify(mockHostListProviderService, times(0)).setInitialConnectionHostSpec(any(HostSpec.class));
   }
+
+  @Test
+  public void testExecuteClearWarnings() throws SQLException {
+    final ReadWriteSplittingPlugin plugin = new ReadWriteSplittingPlugin(
+            mockPluginService,
+            defaultProps,
+            mockHostListProviderService,
+            mockWriterConn,
+            mockReaderConn1);
+
+    plugin.execute(
+            ResultSet.class,
+            SQLException.class,
+            mockStatement,
+            "Connection.clearWarnings",
+            mockSqlFunction,
+            new Object[] {}
+    );
+    verify(mockWriterConn, times(1)).clearWarnings();
+    verify(mockReaderConn1, times(1)).clearWarnings();
+  }
+
+  @Test
+  public void testExecuteClearWarningsOnClosedConnectionsIsNotCalled() throws SQLException {
+    when(mockWriterConn.isClosed()).thenReturn(true);
+    when(mockReaderConn1.isClosed()).thenReturn(true);
+
+    final ReadWriteSplittingPlugin plugin = new ReadWriteSplittingPlugin(
+            mockPluginService,
+            defaultProps,
+            mockHostListProviderService,
+            mockWriterConn,
+            mockReaderConn1);
+
+    plugin.execute(
+            ResultSet.class,
+            SQLException.class,
+            mockStatement,
+            "Connection.clearWarnings",
+            mockSqlFunction,
+            new Object[] {}
+    );
+    verify(mockWriterConn, never()).clearWarnings();
+    verify(mockReaderConn1, never()).clearWarnings();
+  }
+
+  @Test
+  public void testExecuteClearWarningsOnNullConnectionsIsNotCalled() throws SQLException {
```

Comment:
Does this test verify whether `clearWarnings` is or is not called?

Reply:
Good point. Assertion added.